tests/test_gene_association_file.py

```diff
@@ -0,0 +1,151 @@
+import pandas as pd
+import pytest
+from gene_association_file import process_go_annotations
+
+
+@pytest.fixture
+def temp_csv_files(tmp_path):
+    """Create temporary CSV files for input and output testing."""
+
+    input_path = tmp_path / "annotations_data100.csv"
+    output_path = tmp_path / "normalized_annotation.csv"
+
+    example_csv = """DB,DB_Object_ID,DB_Object_Symbol,Qualifier,GO_ID,DB_Reference,Evidence_Code,With_From,Aspect,DB_Object_Name,Synonym,DB_Object_Type,Taxon,Date,Assigned_By,Annotation_Extension,Gene_Product_Form_ID
+UniProtKB,Q04604,TYR,enables,GO:0005507,GO_REF:0000024,ISS,UniProtKB:Q9ZP19,F,Tyrosinase,TYR|TYRS,protein,taxon:8409,20161118,UniProt"""
+
+    input_path.write_text(example_csv)
+    return input_path, output_path
+
+
+@pytest.fixture
+def evidence_mapping_df():
+    """Fixture providing an example mapping DataFrame for testing."""
+    return pd.DataFrame([
+        {"Evidence_Code": "ISS", "DB_Reference": "GO_REF:0000024", "evidence_type": "ECO:0000250"},
+        {"Evidence_Code": "ND", "DB_Reference": "DEFAULT", "evidence_type": "ECO:0000307"},
+    ])
+
+
+@pytest.fixture
+def annotation_df():
+    """Fixture providing an example annotations DataFrame for testing evidence mapping."""
+    return pd.DataFrame([
+        {"Evidence_Code": "ISS", "publications": "GO_REF:0000024"},
+        {"Evidence_Code": "ND", "publications": "PMID:UNKNOWN"},
+    ])
+
+
+@pytest.mark.unit
+def test_go_annotation_processed(tmp_path):
+    """Test that the processed GO annotation CSV file has correct columns."""
+    output_file = tmp_path / "new_annotations1.csv"
+    process_go_annotations("annotations_data100.csv", str(output_file))
+
+    assert output_file.exists(), "Output file not created."
+    df = pd.read_csv(output_file)
+
+    expected_columns = [
+        "subject", "predicate", "object", "publications", "Evidence_Code", "supporting_objects",
+        "annotation_date", "primary_knowledge_source", "aggregator", "protocol_id", "negated", "evidence_type"
+    ]
```

Comment:
this is where it's useful to have documented the col headers in your script 😄 

I usually make a set of constants to represent strings like these so that there's no danger of a typo messing things up -- e.g.

```py
SUBJECT = "subject"
PREDICATE = "predicate"
OBJECT = "object"
...
ASSOCIATION_COL_HEADERS = [SUBJECT, PREDICATE, OBJECT, ... ]
```

It is quite tedious setting up the variables, but it does prevent minor errors from disrupting code.

Reply:
You can also then just import the variable from the script file instead of typing out the expected cols!